confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/ConfigNodeRPCServiceProcessor.java

```diff
@@ -407,6 +410,35 @@ public TSStatus dropFunction(TDropFunctionReq req) throws TException {
     return configManager.dropFunction(req.getUdfName());
   }
 
+  @Override
+  public TSStatus flush(TFlushReq req) throws TException {
+    if (req.storageGroups != null) {
+      List<PartialPath> noExistSg =
+          configManager.checkStorageGroupExist(PartialPath.fromStringList(req.storageGroups));
+      if (!noExistSg.isEmpty()) {
+        StringBuilder sb = new StringBuilder();
+        noExistSg.forEach(storageGroup -> sb.append(storageGroup.getFullPath()).append(","));
+        return RpcUtils.getStatus(
+            TSStatusCode.STORAGE_GROUP_NOT_EXIST,
+            "storageGroup " + sb.subSequence(0, sb.length() - 1).toString() + " does not exist");
+      }
+    }
+
+    List<TDataNodeInfo> onlineDataNodes;
+    if (req.isLocal) {
+      onlineDataNodes = configManager.getNodeManager().getOnlineDataNodes(req.dataNodeId);
+    } else {
+      onlineDataNodes = configManager.getNodeManager().getOnlineDataNodes(-1);
+    }
+    TSStatus tsStatus = new TSStatus();
+    for (TDataNodeInfo dataNodeInfo : onlineDataNodes) {
+      tsStatus =
+          SyncDataNodeClientPool.getInstance()
+              .flush(dataNodeInfo.getLocation().getInternalEndPoint(), req);
+    }
+    return tsStatus;
```

Comment:
It only handle the `status` of last flush request. Failed flush will be ignored if the flushed datanode is not last one in the list.

Reply:
fixed~